add advanced fields calc doc (#4875)
ansys/pyaedt: doc/source/User_guide/pyaedt_extensions_doc/project/advanced_fields_calculator.rst

Advanced Fields Calculator
==========================

The Advanced Fields Calculator provides an additional pool of custom expressions appropriate for each solver.

> These expressions sum up on top of the already existing ones available in the AEDT Fields Calculator.
> These two libraries combined together provide a way to conveniently calculate frequently used quantities.

Reviewer: ```suggestion
These expressions can be used with the already existing ones available in the AEDT Fields Calculator.
These two libraries combined together provide a way to conveniently calculate frequently used quantities.
```
Last sentence, the use of "two libraries" is confusing. Can we change to "two calculators"?

> Operations such as adding, loading and deleting named expressions, as well as creating .clc expressions files, is now

Reviewer: ```suggestion
Operations such as adding, loading, and deleting named expressions and creating CLC expressions files is now
```

> automatized and available in PyAEDT.

Reviewer: ```suggestion
automated and available in PyAEDT.
```

> The core component is a .toml file that functions as an expressions catalog.
> It is located in ``pyaedt>misc>expression_catalog.toml`` and the content for each expression is the following:

Reviewer: ```suggestion
The core component is a TOML file (``expression_catalog.toml``) that functions as an expressions catalog.
It is located in the ``pyaedt>misc`` directory. The content for each expression is the following:
```
I don't understand the last sentence, but I haven't looked at the image.

.. image:: ../../../_static/extensions/expressions_catalog.png
  :width: 800
  :alt: Expressions catalog

> - ``Description``: The name that will be displayed in the UI:

Reviewer: ```suggestion
- ``Description``: Name to display in the UI.
```

.. image:: ../../../_static/extensions/advanced_fields_calc_1.png
  :width: 800
  :alt: Advanced Fields Calculator

- ``Operations``: List of operations to perform to compute the expression.

We tend to divide expressions in ``general`` or ``non-general`` expressions.

> - For general expressions is intended expressions that are independent from a geometry definition.

Reviewer: ```suggestion
General expressions are generally independent of a geometry definition.
```

> In the previous screenshot, to calculate the magnetic field tangential component, there is no need to specify a

Reviewer: ```suggestion
For example, in the previous image, to calculate the magnetic field tangential component, there is no need to specify a
```

geometry assignment.
Whereas in the following example, to calculate the voltage drop along a line, the line assignment is needed for the computation:

.. image:: ../../../_static/extensions/voltage_drop_line.png
  :width: 800
  :alt: Voltage drop line

> To help the user understand this difference we implemented a method called ``is_general_expression(expression_name)``.

Reviewer: ```suggestion
To help you understand this difference, a method named ``is_general_expression(expression_name)`` is implemented.
```

> This returns either True if the expression is general or False otherwise.

Reviewer: ```suggestion
This method returns ``True`` if the expression is general or ``False`` otherwise.
```

It is possible to add named expressions dependent to one another:

.. image:: ../../../_static/extensions/tang_stress_tensor.png
  :width: 800
  :alt: Tangential stress tensor

> There is also the opportunity for the user to add an external .toml file (that must be placed in PersonalLib folder)
> to load custom expressions. This could be especially useful if the user does not want to share expressions.
> To load a personalized .toml file use ``load_expression_file(toml_file_path)``.

Reviewer: ```suggestion
It is also possible for you to add an external TOML file in the ``PersonalLib`` folder
to load custom expressions. This could be especially useful if you do not want to share expressions.
To load a personalized TOML file, use the ``load_expression_file(toml_file_path)`` method.
```
